disnake/ui/button.py

```diff
@@ -304,20 +352,18 @@ def button(
         ordering. The row number must be between 0 and 4 (i.e. zero indexed).
     """
 
-    def decorator(func: ItemCallbackType[Button]) -> DecoratedItem[Button]:
+    if (origin := get_origin(cls)) is not None:
```

Comment:
The generic parameter on `cls` (if any) should ideally be preserved, `get_origin` is only necessary for the following type check

Reply:
I'm not sure what do you mean. I'm pretty sure that if you specify `cls=CustomButton[int]`, the generic type is reflected by the decorator? Except when you don't specify the `cls`, in which case it inherits the more specific type from the callback annotation

Reply:
Ah yes, nevermind that. I was thinking of accessing the generic parameter at runtime, but there isn't really a point in doing that and `_BaseGenericAlias.__call__` already strips it when calling the item's `__init__`, so it wouldn't matter in most cases anyhow.